src/Tests/Microsoft.DotNet.ApiCompatibility.Tests/Rules/CannotAddOrRemoveVirtualModifierTests.cs

```diff
@@ -0,0 +1,115 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.DotNet.ApiCompatibility.Abstractions;
+using Xunit;
+
+namespace Microsoft.DotNet.ApiCompatibility.Tests
+{
+    public class CannotAddOrRemoveVirtualModifierTests
+    {
+        [Fact]
+        public static void DifferencesReported()
+        {
+            string leftSyntax = @"
+namespace CompatTests
+{
+  public abstract class First {
+    public virtual void F() {}
+    public void G() {}
+    public abstract void H();
+  }
+}
+";
+            string rightSyntax = @"
+namespace CompatTests
+{
+  public abstract class First {
+    public void F() {}
+    public virtual void G() {}
+    public virtual void H() {}
+  }
+}
+";
+            IAssemblySymbol left = SymbolFactory.GetAssemblyFromSyntax(leftSyntax);
+            IAssemblySymbol right = SymbolFactory.GetAssemblyFromSyntax(rightSyntax);
+            ApiComparer differ = new();
+            IEnumerable<CompatDifference> differences = differ.GetDifferences(new[] { left }, new[] { right });
+            CompatDifference[] expected = new[]
+            {
+                new CompatDifference(DiagnosticIds.CannotRemoveVirtualModifier, string.Empty, DifferenceType.Removed, "M:CompatTests.First.F"),
+                new CompatDifference(DiagnosticIds.CannotAddVirtualModifier, string.Empty, DifferenceType.Added, "M:CompatTests.First.G"),
+            };
+            Assert.Equal(expected, differences);
+        }
+
+        [Fact]
+        public static void Null()
```

Comment:
Can we change the test names to names that are more descriptive of what is being tested? For example, for this one, I would write something like: `EnsureNoDiagnosticsLoggedForAddedOrRemovedVirtualMembers()`